### PR TITLE
Fixes part of #915: App language low fi [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by rt4914**
## Explanation

Fixes part of #915 

This PR makes sure that app language change from Options menu is reflected in entire app wherever we are using strings.xml file on per profile basis.

## NOTE:
This is just for checking the implementation and complete PR review but it will not get merged for until and unless alpha is ready because we will need to update the translations for `strings.xml` as new strings will get introduced in other PRs.

## Points to keep in mind while testing
1. Go to `OptionActivity` and change language to `Hindi`. Currently we have support of hindi only.
2. See that this change is reflected in all activities which appear after `HomeActivity`.
3. ProfileChooser, AdminAuth, AddProfile all these activities should not show language change as this language change is completely profile dependent.
4. For one profile set `English` and for other set `Hindi`. And check that works correctly.

## Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by nikitamarysolomanpvt** [comment]
@rt4914  Please add screen shots.

File: app/src/main/java/org/oppia/app/home/HomeActivityPresenter.kt [hunk 1]
```diff
@@ -43,4 +55,53 @@ class HomeActivityPresenter @Inject constructor(private val activity: AppCompatA
...
+
+  private fun subscribeToProfileLiveData() {
+    getProfileData().observe(activity, Observer<Profile> {
+      setPreferredLanguage(getAppLanguageCode(it.appLanguage))
+    })
+  }
+
+  private fun processGetProfileResult(profileResult: AsyncResult<Profile>): Profile {
+    if (profileResult.isFailure()) {
+      logger.e("HomeActivity", "Failed to retrieve profile", profileResult.getErrorOrNull()!!)
```
Comment (nikitamarysolomanpvt) [inline app/src/main/java/org/oppia/app/home/HomeActivityPresenter.kt:71]: Please add "HomeActivity" as constant tag

File: app/src/main/java/org/oppia/app/profile/ProfileActivityPresenter.kt [hunk 2]
```diff
@@ -41,4 +44,15 @@ class ProfileActivityPresenter @Inject constructor(
   private fun getProfileChooserFragment(): ProfileChooserFragment? {
     return activity.supportFragmentManager.findFragmentById(R.id.profile_chooser_fragment_placeholder) as ProfileChooserFragment?
   }
+
+  private fun setPreferredLanguage() {
```
Comment (nikitamarysolomanpvt) [inline app/src/main/java/org/oppia/app/profile/ProfileActivityPresenter.kt:48]: Please add kdoc for this method.

**Comment by rt4914** [comment]
Screenshot_1585648812

Actually the steps of testing this have been mentioned in the description. At the same time it deals with language change which can be seen for strings.xml lines across various activities.

**Review by nikitamarysolomanpvt — APPROVED** [review]
Some nit changes  beside that LGTM

**Review by veena14cs — APPROVED** [review]
LGTM.

**Comment by BenHenning** [comment]
Since this hasn't seen progress in a while, closing it. We can still use it as a reference, or reopen if we want to continue contributing to this branch.

**Pull request closed without merging**